accepted/0000-workspace-run-commands.md

```diff
@@ -0,0 +1,67 @@
+- Start Date: 2017-09-21
+- RFC PR:
+- Yarn Issue:
+
+# Summary
+
+Allow Yarn CLI to execute a package script on each packages from the workspace root.  
+
+# Motivation  
+
+"Lerna". Lerna does a great job for handling monorepos. Since Yarn has a built-in 
+workspaces feature, we could use some of the functionalities that being used in 
+lerna (not entirely though!) for handling the monorepo more efficiently.  
+
+Just like installing all dependencies from one place (workspace root), it'd be great 
+to execute all the sub-package scripts from the workspace root using a single command. 
+Not only this would make a less back-and-forth traveling between the packages to 
+execute scripts, but also it'll help a lot for CI/CD configuration. 
+
+
+# Detailed design  
+
+As @BYK mentioned [here](https://github.com/yarnpkg/yarn/issues/4467#issuecomment-330873337), 
+we could create two commands for this feature.  
+
+* `yarn workspaces list`  
+* `yarn workspaces run <command>`  
+* `yarn workspaces add <workspace_name>`  
+* `yarn workspace remove <workspace_name> --opts`
+
+By introducing `workpaces` command, it brings a modular approach from the CLI's perspective.  
+
+## `yarn workspaces list`  
+
+This command lists out the package names inside that workspace.  In the case of multiple workspace configuration, we'll need to pass the workspace name like `yarn workspaces <workspace_name> list`. By defaut, when no workspace name is not given, it'll pick the first workspace name from the root package.json.  
+
+## `yarn workspaces run <command>`  
+
+This command will execute the specified command in all the packages under the workspace. Before we run the scripts, it should make sure that all packages has the script specified in the package.json. Otherwise, it should throw an error specifying which pacakge is missing that script.  
+
+Just like the `list` command, by default, it'll select the first workspace in case of no workspace name is provided.  
```

Comment:
I think this should again run the command in all workspaces, not just the first one. And then later a "filtering" mechanism can be added to restrict further. But otherwise it seems to arbitrary.

Reply:
I think, like @bestander said earlier, if we want to execute a command in a specific workspace, it's better to `cd` to that package and `yarn run`. 

I've updated this command to execute all in all workspaces.